### PR TITLE
CI: Basic Performance check reporting

### PR DESCRIPTION
**Pull request by PeterTh**
This PR adds a CI action which checks the microbenchmark results, as discussed at the last in-person meeting.

- [x] It generates a warning if the benchmark file was not updated.
- [x] It reports any significant deviations.
- [x] It generates and uploads box plots of all the benchmark results.
- [x] Box plots with significant changes are marked (red border).
- [x] If a new benchmark is added, or a benchmark is removed, this is reported.
- [x] A new comment is only added if the benchmark csv file changed since the previous version in the same PR.

### CODE REVIEW

**Comment by fknorr** [comment]
Very neat! Do we have any historic PRs where we significantly improved / regressed performance to calibrate the THRESHOLDs?

**Comment by PeterTh** [comment]
This is now done and fully functional, as far as I can tell.

**Comment by fknorr** [comment]
I'm pretty happy with this! Two minor things:

- Could we have labels, or at least units on the y axis? Even though the scale doesn't matter, it's useful to see whether we're comparing rates or times.
- We're running all (?) other CI jobs on our own runners, should we do the same for performance reporting? We could then simply include the relevant ruby environment in our docker images and improve CI performance a bit.

**Comment by PeterTh** [comment]
Sure, I can add the unit on the Y axis. I believe for all these benchmarks it's currently simply execution time in nanoseconds.

I purposefully didn't run this on our own runners since it has no dependencies on our own infrastructure, and so it made more sense to me not to add another dependency to our own stuff. 
I'm not sure what you mean in terms of CI performance. Since this runs on GitHub runners, it starts at the same time as everything else, and it takes less than a minute. So it will always be done long before everything else is ready.

**Comment by PeterTh** [comment]
There is now a unit indicator on the Y axis.

**Review by fknorr — APPROVED** [review]

File: ci/check-perf-impact.rb [hunk 1]
```diff
@@ -0,0 +1,248 @@
+require 'csv'
+require 'gruff'
+require 'json'
+require 'digest'
+
+# information regarding the benchmark file
+BENCH_FN = 'ci/perf/gpuc2_bench.csv'
+MEAN_COL = 5       # column containing the mean value
```
Comment (psalz) [inline ci/check-perf-impact.rb:8]: Unused

**Review by psalz — APPROVED** [review]
LGTM!

File: ci/check-perf-impact.rb (hunk 1 above)
Comment (PeterTh) [reply ci/check-perf-impact.rb]: Thanks, removed.

**Pull request merged**